lndr-backend/src/Lndr/Server.hs

```diff
@@ -228,19 +230,28 @@ verifySettlementsWithTxHash :: LndrHandler ()
 verifySettlementsWithTxHash = do
     (ServerState pool configTVar _) <- ask
     config <- liftIO $ readTVarIO configTVar
-    creditHashes <- liftIO $ withResource pool Db.settlementCreditsToVerify
-    mapM_ verifyIndividualRecord creditHashes
+    txHashes <- liftIO $ withResource pool Db.txHashesToVerify
+    creditsToVerify <- mapM (\txHash -> liftIO $ withResource pool $ Db.lookupCreditsByTxHash txHash) txHashes
+    mapM_ verifyRecords creditsToVerify
 
 
-verifyIndividualRecord :: TransactionHash -> LndrHandler ()
-verifyIndividualRecord creditHash = do
+verifyRecords :: [BilateralCreditRecord] -> LndrHandler ()
+verifyRecords records = do
     (ServerState pool configTVar loggerSet) <- ask
-    recordM <- liftIO $ withResource pool $ Db.lookupCreditByHash creditHash
-    let recordNotFound = throwError $
-            err404 { errBody = "Credit hash does not refer to pending bilateral settlement record" }
-    bilateralCreditRecord <- maybe recordNotFound pure recordM
-    verifySettlementPayment bilateralCreditRecord
-    liftIO $ withResource pool $ Db.verifyCreditByHash creditHash
-    web3Result <- finalizeTransaction configTVar bilateralCreditRecord
-                        `catchError` (pure . T.pack . show)
+    
+    initialTxHash <- maybe (throwError (err500 {errBody = "Bilateral Settlement Record does not have txHash."})) 
+                     pure . txHash $ head records
+    -- this should only take the creditor, debtor, credit hash, and settlement amount
+    let firstCreditor = creditor $ creditRecord $ head records
+        firstDebtor = debtor $ creditRecord $ head records
+        creditorAmount = sum . fmap (fromMaybe 0 . settlementAmount . creditRecord) $ filter ((== firstCreditor) . creditor . creditRecord) records
+        debtorAmount = sum . fmap (fromMaybe 0 . settlementAmount . creditRecord) $ filter ((/= firstCreditor) . creditor . creditRecord) records
+    
+    when (creditorAmount > debtorAmount) $
```

Comment:
not good form to have two mutually exclusive `when`s in a row. just calculate the absolute value of settlementAmount `(abs $ creditorAmount - debtorAmount)` since we should never have a negative value for this, and assign creditor / debtor in a let statement above using an if statement.